csharp/src/Ice/ConnectionI.cs

```diff
@@ -2516,6 +2532,12 @@ private void sendResponse(OutgoingResponse response, bool isTwoWay, byte compres
                         sendMessage(new OutgoingMessage(response.outputStream, compress > 0, adopt: true));
                     }
 
+                    if (_maxDispatches > 0 && _state != StateHolding && _dispatchCount == _maxDispatches)
```

Comment:
I am not following here. The _maxDispatches is "active" only state = StateHolding? Should it be only when state = StateActive?

Reply:
Did you expect `_state == StateActive` instead? It's pretty much the same.